### PR TITLE
Url pagination

### PR DESCRIPTION
**Pull request by Racenna**

### CODE REVIEW

File: src/components/Users/UsersContainer.jsx [hunk 1]
```diff
@@ -21,14 +23,27 @@ const UsersContainer = () => {
...
+  const history = useHistory();
 
   const onPageChanged = (page) => {
     dispatch(getUsers(page, pageSize));
   };
 
   useEffect(() => {
-    dispatch(getUsers(currentPage, pageSize));
-  }, [currentPage, pageSize, dispatch]);
+    const parsed = queryString.parse(history.location.search);
```
Comment (worksnk) [inline src/components/Users/UsersContainer.jsx:33]: knowing about queryString is good. But I want you to use build capabilities to work with query

Also I'm not sure why you use history if you need location. Just use location hook directly

File: src/components/Users/UsersContainer.jsx [hunk 2]
```diff
@@ -21,14 +23,27 @@ const UsersContainer = () => {
...
+
+    let actualPage = currentPage;
+
+    if (!!parsed.page) actualPage = Number(parsed.page);
+    dispatch(getUsers(actualPage, pageSize));
+  }, []);
+
+  useEffect(() => {
```
Comment (worksnk) [inline src/components/Users/UsersContainer.jsx:41]: Why do you trigger it on each render of component?

Also change it to be declarative

File: src/components/Users/UsersContainer.jsx [hunk 3]
```diff
@@ -21,14 +23,27 @@ const UsersContainer = () => {
...
 
   useEffect(() => {
-    dispatch(getUsers(currentPage, pageSize));
-  }, [currentPage, pageSize, dispatch]);
+    const parsed = queryString.parse(history.location.search);
+
+    let actualPage = currentPage;
+
+    if (!!parsed.page) actualPage = Number(parsed.page);
+    dispatch(getUsers(actualPage, pageSize));
```
Comment (worksnk) [inline src/components/Users/UsersContainer.jsx:38]: In general this is incorrect you want refetch when currentPage is changed and that's it

**Review by worksnk — CHANGES_REQUESTED** [review]

File: src/components/Users/UsersContainer.jsx (hunk 1 above)
Comment (Racenna) [reply src/components/Users/UsersContainer.jsx]: replaced history with location

File: src/components/Users/UsersContainer.jsx (hunk 2 above)
Comment (Racenna) [reply src/components/Users/UsersContainer.jsx]: added dependecy in useEffect

File: src/redux/usersReducer/selectors.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export const getCurrentPageSelector = (state) => state.usersData.currentPage;
```
Comment (worksnk) [inline src/redux/usersReducer/selectors.js:1]: Don't store it redux

File: src/redux/profileReducer/reducer.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+const profileReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_POST:
+      return {
+        ...state,
+        posts: [
+          ...state.posts,
+          {
+            id: 65,
```
Comment (worksnk) [inline src/redux/profileReducer/reducer.js:25]: why you put always the same id?

File: src/components/Users/UsersContainer.jsx [hunk 6]
```diff
@@ -24,26 +23,25 @@ const UsersContainer = () => {
...
-    const parsed = queryString.parse(history.location.search);
-
-    let actualPage = currentPage;
-
-    if (!!parsed.page) actualPage = Number(parsed.page);
-    dispatch(getUsers(actualPage, pageSize));
-  }, []);
-
-  useEffect(() => {
     history.push({
```
Comment (worksnk) [inline src/components/Users/UsersContainer.jsx:30]: Don't use history push. It must redirect to url when clicked. Declrative

**Review by worksnk — APPROVED** [review]

**Pull request merged**